client/components/EventsPlanningFilters/EditFilterSchedule.tsx

```diff
@@ -41,88 +49,167 @@ export class EditFilterScheduleComponent extends React.Component<IProps, IState>
         this.onSaveHandler = this.onSaveHandler.bind(this);
         this.previewFilter = this.previewFilter.bind(this);
         this.onChange = this.onChange.bind(this);
+        this.updateHour = this.updateHour.bind(this);
+        this.addHour = this.addHour.bind(this);
+        this.removeHour = this.removeHour.bind(this);
     }
 
     onSaveHandler() {
-        const schedule = Object.assign({}, this.state.schedule);
+        const schedule = {...this.state.schedule};
 
-        switch (schedule.frequency) {
-        case SCHEDULE_FREQUENCY.HOURLY:
-            delete schedule.hour;
-            delete schedule.day;
-            delete schedule.week_days;
-            break;
-        case SCHEDULE_FREQUENCY.WEEKLY:
-            delete schedule.day;
-            if (!schedule.week_days?.length) {
-                schedule.week_days = [
-                    WEEK_DAY.SUNDAY,
-                    WEEK_DAY.MONDAY,
-                    WEEK_DAY.TUESDAY,
-                    WEEK_DAY.WEDNESDAY,
-                    WEEK_DAY.THURSDAY,
-                    WEEK_DAY.FRIDAY,
-                    WEEK_DAY.SATURDAY,
-                ];
-            }
-            break;
-        case SCHEDULE_FREQUENCY.MONTHLY:
-            delete schedule.week_days;
-            break;
+        // Clean up legacy `hour` field
+        delete schedule.hour;
```

Comment:
Instead of leaving a deprecated field on the data, we can instead use a migration script to update the existing data so we won't need code to support that old data format.